include/bluetooth/bluetooth.h

```diff
@@ -45,6 +45,51 @@ extern "C" {
  */
 #define BT_ID_DEFAULT 0
 
+/** Opaque type representing a connection to a remote device */
+struct bt_adv;
+
+/* Don't require everyone to include conn.h */
+struct bt_conn;
+
+struct bt_le_adv_cb {
+	/** @brief The advertising set has finished sending adv data.
+	 *
+	 *  This callback notifies the application that the advertising set has
+	 *  finished sending advertising data.
+	 *  The advertising set can either have been stopped by a timeout or
+	 *  because because the specified number of advertising intervals has
+	 *  been reached.
+	 *
+	 *  @param adv The advertising set object.
+	 *  @param num_sent The number of advertising events completed.
+	 */
+	void (*sent)(struct bt_adv *adv, u8_t num_sent);
+
+	/** @brief The advertising set has accepted a new connection.
+	 *
+	 *  This callback notifies the application that the advertising set has
+	 *  accepted a new connection.
+	 *
+	 *  @param adv The advertising set object.
+	 *  @param num_sent The new connection object.
+	 */
+	void (*connected)(struct bt_adv *adv, struct bt_conn *conn);
```

Comment:
on HCI level what is here  sent and connected callback is single event,  why split this?  eg num_sent is also valid when set was terminated by connection

Reply:
I thought it was a reasonable simplification. Since they are two different things.
If the application is interested in both, then both callbacks will be called.

Reply:
If the application set num_sent, then sent callback will be called when adv set is terminated.
If the adv set is terminated because it was connected, then connected will be called.

So both could happen, or just one.